pkg/image/image.go

```diff
@@ -92,6 +92,7 @@ func ParseReference(ref string) (TypedImageReference, error) {
 
 	dstType := DestinationOCI
 
+	// TODO: this assumes you can parse it into a docker image ref (and you can’t do that since its just a path).
```

Comment:
I get what you're saying... DockerImageReference has to be provided, but it is not really a valid docker reference for an OCI path. The DockerImageReference is constructed with the values that come from the result of `v1alpha2.ParseImageReference`, but this content is going to be variable depending on the path that's feed into `v1alpha2.ParseImageReference`.

In my next push, there's a test case called `TestV1A2ParseImageReferenceOCIRefs` in `image_test.go` which covers all of the bases of what `v1alpha2.ParseImageReference` does with an OCI path (or at least I think it does).

Since `image.ParseImageReference` function is called all over the place, it's hard to talk about specifically, but one example is in create.go in the `run` function. It iterates over the `[]v1alpha2.Operator` slice, and uses the resulting `parsedRef.Ref.Name` as an image reference. If you look in the test case I mentioned, that equates to the `expRepo` field in the test case values. Essentially the last segment of the path is used as the reference. In some edge cases, the parsing results is an empty string (e.g. if you've been given `/` as the path or if the path is relative without any directory name provided). These edge cases are likely to result in an error (and maybe that's OK).

To flesh out what I am talking about:
1. Start with "oci:///Users/jhunkins/temp/ocmirror" as a reference
2. This becomes "ocmirror"
3. Now "ocmirror" gets converted into a go container registry reference and stored in `CatalogMetadata.CatalogRef`. That effectively looks something like this:

```
index.docker.io/library/ocmirror@sha256:dc1b9846d7994450e74b9cde2e621f8c1d98cdf1debd591db291785dd3fc6446
```

In the end we have various ways to prevent `CatalogMetadata.CatalogRef` from being used as a real image reference. We just need to be vigilant to not accidentally use `image.ParseImageReference` results in ways that don't take into account the fact that it could be a bogus reference for OCI refs. My comment was intended to drive this conversation. I can change the comment to just be a warning that states this is generating a bogus docker image reference for OCI and leave it at that.